### PR TITLE
MNT Use faulthandler rather than custom autokill logic

### PR DESCRIPTION
**Pull request by lesteve**
This should give more information in case of deadlock.

### CODE REVIEW

**Review by ogrisel — APPROVED** [review]
Assuming greenish CI, +1 on my side :)

File: joblib/test/common.py [hunk 1]
```diff
@@ -62,43 +62,6 @@ def dummy_func():
 # triggers an infinite wait on a pipe by the master process for one of its
 # failed workers
```
Comment (ogrisel) [inline joblib/test/common.py:63]: The comment above should be deleted as well.

**Review by ogrisel — APPROVED** [review]
LGTM with this comment.

**Review by ogrisel — APPROVED** [review]

**Review by tomMoral — APPROVED** [review]
LGTM! thx @lesteve

**Pull request merged**